spec/services/import_polygons_spec.rb

```diff
@@ -2,84 +2,151 @@
-
-      it "imports Bath city" do
-        expect(LocationPolygon.cities.first.name).to eq("bath")
+    describe "#get_buffers" do
+      let(:buffer_response) { JSON.parse(file_fixture("buffer_response.json").read) }
+      let(:buffer_coordinates) { [65.06414131400004, -12.5761721229999353, 65.06185937400005, -12.5761901179999427] }
+      let(:imported_buffers) do
+        {
+          "5" => buffer_coordinates,
+          "10" => buffer_coordinates,
+          "15" => buffer_coordinates,
+          "20" => buffer_coordinates,
+          "25" => buffer_coordinates,
+        }
       end
 
-      it "does not import non-existent city" do
-        expect(LocationPolygon.cities.count).to eq(1)
+      context "when api_endpoint is shorter than URL_MAXIMUM_LENGTH" do
+        before do
+          distances = [5, 10, 15, 20, 25]
+          distances.each do |distance|
+            allow(HTTParty).to receive(:get).with(
+              "https://ons-inspire.esriuk.com/arcgis/rest/services/Utilities/Geometry/GeometryServer/buffer?"\
+              "bufferSR=3857&distances=#{convert_miles_to_metres(distance)}&f=json&geodesic=f"\
+              "alse&geometries=%7B%22geometryType%22%3D%3E%22esriGeometryPolygon%22%2C+%22geometries%22%3D%3E"\
+              "%5B%7B%22rings%22%3D%3E%5B%5B%5B55.8110853660943%2C+-2.0343575091738%5D%2C+%5B55.7647624900862%"\
+              "2C+-1.9841097397706%5D%5D%5D%7D%5D%7D&inSR=4326&outSR=4326&unionResults=true&unit=",
+            ).and_return(buffer_response)
+          end
+          subject.call
+        end
+
+        let(:location_type) { :regions }
+        let(:file_name) { "ons_regions.json" }
+
+        it "imports buffers" do
+          expect(LocationPolygon.regions.first.buffers).to eq(imported_buffers)
+        end
       end
 
-      it "imports Bath boundaries" do
-        expect(LocationPolygon.cities.first.boundary).to eq(points)
+      context "when api_endpoint is longer than URL_MAXIMUM_LENGTH" do
+        let(:file_name) { "ons_counties_response_with_more_coords.json" }
+        let(:location_type) { :counties }
+
+        before do
+          stub_const("ImportPolygons::URL_MAXIMUM_LENGTH", 400)
+          distances = [5, 10, 15, 20, 25]
+          distances.each do |distance|
+            allow(HTTParty).to receive(:get).with(
+              "https://ons-inspire.esriuk.com/arcgis/rest/services/Utilities/Geometry/GeometryServer/buffer?"\
+              "bufferSR=3857&distances=#{convert_miles_to_metres(distance)}&f"\
+              "=json&geodesic=false&geometries=%7B%22geometryType%22%3D%3E%22esriGeometryPolygon%22%2C+%22"\
+              "geometries%22%3D%3E%5B%7B%22rings%22%3D%3E%5B%5B%5B52%2C+0%5D%2C+%5B52%2C+0%5D%5D%5D%7D%5D%7D"\
+              "&inSR=4326&outSR=4326&unionResults=true&unit=",
+            ).and_return(buffer_response)
+          end
+          subject.call
+        end
+
+        it "reduces the number of coordinates before constructing the API endpoint" do
```

Comment:
Re-reading this description, it feels too unrelated to what's happening on line 148. Someone could read it and think 'how does checking that the buffers are there prove that the coordinates got reduced'?

I suggest:

```ruby
it "is still able to import the buffers, by reducing the number of coordinates in the params to the API endpoint" do
```